core/drivers/crypto/se050/core/storage.c

```diff
@@ -5,21 +5,40 @@
  */
 
 #include <crypto/crypto.h>
+#include <mm/core_mmu.h>
 #include <se050.h>
 #include <se050_utils.h>
 #include <string.h>
+#include <tee/tee_fs.h>
+#include <tee/tee_obj.h>
+#include <tee/tee_pobj.h>
 
-void crypto_storage_obj_del(uint8_t *data, size_t len)
+void crypto_storage_obj_del(struct tee_obj *o)
 {
 	sss_status_t status = kStatus_SSS_Success;
 	uint32_t val = SE050_KEY_WATERMARK;
 	sss_se05x_object_t k_object = { };
+	uint8_t *data = NULL;
 	bool found = false;
-	uint8_t *p = data;
+	size_t len = 0;
+	uint8_t *p;
 
-	if (!p)
+	if (!o)
 		return;
 
+	len = o->info.dataSize;
+	data = calloc(1, len);
```

Comment:
Such failure will silebntly not free data. This might be an issue!
Should return OUT_OF_MEMORY.